x-pack/legacy/plugins/apm/public/components/app/Settings/AgentConfigurations/AgentConfigurationCreateEdit/index.stories.tsx

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { storiesOf } from '@storybook/react';
+import React from 'react';
+import { HttpSetup } from 'kibana/public';
+import { createCallApmApi } from '../../../../../services/rest/createCallApmApi';
+import { AgentConfigurationCreateEdit } from './index';
+import {
+  ApmPluginContext,
+  ApmPluginContextValue
+} from '../../../../../context/ApmPluginContext';
+
+storiesOf(
+  'app/Settings/AgentConfigurations/AgentConfigurationCreateEdit',
+  module
+).add(
+  'with config',
+  () => {
+    const httpMock = {};
+
+    // mock
+    createCallApmApi((httpMock as unknown) as HttpSetup);
+
+    const contextMock = {
+      core: {
+        notifications: { toasts: { addWarning: () => {}, addDanger: () => {} } }
+      }
+    };
```

Comment:
Should we share these mock, so they don't need to be recreated for every story?

I think a `core` mock might be too big but a mock per module, eg. `notificationsMock`, `httpMock` etc.

Reply:
I like what we have on testHelper (https://github.com/sqren/kibana/blob/agent-configuration-additional-settings/x-pack/legacy/plugins/apm/public/utils/testHelpers.tsx#L222) a wrapper which takes care of everything if needed the values can be overwritten

Reply:
I did at one point have trouble importing that mock into a story (it might be because of how the storybook setup's webpack config works), but if you can, I would recommend using the mocks from testHelpers. You should be able to override some values if you need to.